cloudca/resource_cloudca_vpn.go

```diff
@@ -0,0 +1,151 @@
+package cloudca
+
+import (
+	"fmt"
+
+	"github.com/cloud-ca/go-cloudca"
+	"github.com/hashicorp/terraform/helper/schema"
+)
+
+func resourceCloudcaVpn() *schema.Resource {
+	return &schema.Resource{
+		Create: resourceCloudcaVpnCreate,
+		Read:   resourceCloudcaVpnRead,
+		Delete: resourceCloudcaVpnDelete,
+
+		Importer: &schema.ResourceImporter{
+			State: schema.ImportStatePassthrough,
+		},
+
+		Schema: map[string]*schema.Schema{
+			"environment_id": {
+				Type:        schema.TypeString,
+				Required:    true,
+				ForceNew:    true,
+				Description: "ID of the environment where the vpn should be created",
+			},
+			"vpc_id": {
+				Type:        schema.TypeString,
+				Required:    true,
+				ForceNew:    true,
+				Description: "Id of the VPC for the vpn",
+			},
+			"certificate": {
+				Type:        schema.TypeString,
+				ForceNew:    true,
+				Computed:    true,
+				Description: "Certificate to use when using IKEV2 vpn type",
+			},
+			"preshared_key": {
+				Type:        schema.TypeString,
+				ForceNew:    true,
+				Computed:    true,
+				Description: "Preshared key to use when using L2TP vpn type",
+			},
+			"public_ip": {
+				Type:        schema.TypeString,
+				ForceNew:    true,
+				Computed:    true,
+				Description: "Public IP address associated with the vpn",
+			},
+			"public_ip_id": {
+				Type:        schema.TypeString,
+				ForceNew:    true,
+				Computed:    true,
+				Description: "ID of the public IP address associated with the vpn",
+			},
+			"state": {
+				Type:        schema.TypeString,
+				ForceNew:    true,
+				Computed:    true,
+				Description: "State of the vpn",
+			},
+			"type": {
+				Type:        schema.TypeString,
+				ForceNew:    true,
+				Computed:    true,
+				Description: "Type of vpn connection",
+			},
+		},
+	}
+}
+
+func resourceCloudcaVpnCreate(d *schema.ResourceData, meta interface{}) error {
+	vpnIPPurpose := "SOURCE_NAT"
+	ccaResources, rerr := getResourcesForEnvironmentID(meta.(*cca.CcaClient), d.Get("environment_id").(string))
+	if rerr != nil {
+		return rerr
+	}
+
+	var vpnPubIPID string
+	pubIps, _ := ccaResources.PublicIps.List()
+	for _, ip := range pubIps {
+		if ip.VpcId == d.Get("vpc_id").(string) {
+			for _, purpose := range ip.Purposes {
+				if purpose == vpnIPPurpose {
+					vpnPubIPID = ip.Id
```

Comment:
You can break out of both for loops here

Reply:
Fixed...